internal/controller/nnf_clientmount_controller.go

```diff
@@ -282,6 +304,50 @@ func (r *NnfClientMountReconciler) changeMount(ctx context.Context, clientMount
 	return nil
 }
 
+// Retrieve the Servers resource for the workflow and write it to a dotfile on the mount path for compute users to retrieve
+func (r *NnfClientMountReconciler) dumpServersToFile(ctx context.Context, clientMount *dwsv1alpha2.ClientMount, path string) error {
+	// Obtain the correct workflow/namespace from the Clientmount's labels
+	wf, wfFound := clientMount.Labels[dwsv1alpha2.WorkflowNameLabel]
+	ns, wfNsFound := clientMount.Labels[dwsv1alpha2.WorkflowNamespaceLabel]
+	if !wfFound || !wfNsFound {
+		return dwsv1alpha2.NewResourceError("unable to determine workflow from labels").WithMajor()
+	}
+
+	// Use the workflow/namespace labels to find the matching Server resource
+	matchLabels := dwsv1alpha2.MatchingWorkflow(&dwsv1alpha2.Workflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      wf,
+			Namespace: ns,
+		},
+	})
+	listOptions := []client.ListOption{
+		matchLabels,
```

Comment:
You'll also need to make use of the `nnf.cray.hpe.com/directive_index` label here. If there are multiple jobdw #DWs there will be multiple Servers resource with the same Workflow labels